android/src/main/java/com/urbanairship/reactnative/EventEmitter.java

```diff
@@ -114,22 +115,46 @@ public void onReactContextInitialized(ReactContext reactContext) {
     }
 
     /**
-     * Helper method to add known listeners.
+     * Called when a new listener is added for a specified event name.
      *
-     * @param eventName The event name identifying the listener.
+     * @param eventName The event name.
      */
-    void addKnownListener(String eventName) {
-        knownListeners.add(eventName);
+    void addAndroidListener(ReactContext reactContext, String eventName) {
+        synchronized (knownListeners) {
+            List<Event> pending = pendingEvents;
```

Comment:
pretty sure this will cause an error. you need to copy the list, not just point a different variable to it